redaxo/src/addons/media_manager/lib/effects/effect_convert2img.php

```diff
@@ -85,18 +83,15 @@ public function execute()
             return false;
         }
 
-        $image_src = call_user_func($convert_to['createfunc'], $to_path);
-
-        if (!$image_src) {
-            return;
-        }
-
-        $this->media->setImage($image_src);
+        $this->media->setSourcePath($to_path);
         $this->media->refreshImageDimensions();
         $this->media->setFormat($convert_to['ext']);
         $this->media->setMediaFilename($filename);
         $this->media->setHeader('Content-Type', $convert_to['content-type']);
-        unlink($to_path);
+
+        register_shutdown_function(function () use ($to_path) {
+            rex_file::delete($to_path);
+        });
```

Comment:
Beim Testen war mir übrigens aufgefallen, dass man dann das Problem hat, dass man ja die temporäre Datei auch noch wieder löschen möchte. Das kann man aber erst nach Auslieferung machen.
Habe es hier mal in convert2img demonstriert.
@staabm Ist der Weg über eine Shutdown-Function hier ok?